docs/pyproject.toml

```diff
@@ -5,4 +5,4 @@ line-length = 74
 
 # Always use our latest supported version here since we want code snippets in
 # docs to use the most up-to-date syntax.
-target-version = ['py312']
+target-version = ['py39']
```

Comment:
Yes but also our black version doesn't support 3.12 and the formatter / linter doesn't run locally anymore:

```
❯ docker-compose run baseplate make fmt
[+] Creating 5/0
 ✔ Container baseplatepy-memcached-1           Running                                                                                                                                0.0s
 ✔ Container baseplatepy-zookeeper-1           Running                                                                                                                                0.0s
 ✔ Container baseplatepy-redis-cluster-node-1  Running                                                                                                                                0.0s
 ✔ Container baseplatepy-redis-1               Running                                                                                                                                0.0s
 ✔ Container baseplatepy-cassandra-1           Running                                                                                                                                0.0s
...
black baseplate/ tests/
Usage: black [OPTIONS] SRC ...
Try 'black -h' for help.

Error: Invalid value for '-t' / '--target-version': invalid choice: py312. (choose from py27, py33, py34, py35, py36, py37, py38, py39)
make: *** [Makefile:55: fmt] Error 2
```

Reply:
Yes, but also we need to upgrade black because it appears to not work with py312:

```
❯ docker-compose run baseplate make fmt
[+] Creating 6/6
 ✔ Container baseplatepy-cassandra-1                                                                                                                                 Created          0.1s
 ✔ Container baseplatepy-redis-1                                                                                                                                     Created          0.1s
 ✔ Container baseplatepy-zookeeper-1                                                                                                                                 Created          0.1s
 ✔ Container baseplatepy-redis-cluster-node-1                                                                                                                        Created          0.1s
 ✔ Container baseplatepy-memcached-1                                                                                                                                 Created          0.1s
 ! redis-cluster-node The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested                  0.0s
[+] Running 5/5
 ✔ Container baseplatepy-cassandra-1           Started                                                                                                                                0.3s
 ✔ Container baseplatepy-memcached-1           Started                                                                                                                                0.4s
 ✔ Container baseplatepy-zookeeper-1           Started                                                                                                                                0.4s
 ✔ Container baseplatepy-redis-cluster-node-1  Started                                                                                                                                0.4s
 ✔ Container baseplatepy-redis-1               Started                                                                                                                                0.3s
...
black baseplate/ tests/
Usage: black [OPTIONS] SRC ...
Try 'black -h' for help.

Error: Invalid value for '-t' / '--target-version': invalid choice: py312. (choose from py27, py33, py34, py35, py36, py37, py38, py39)
make: *** [Makefile:55: fmt] Error 2
```